app/tests/test_search.py

```diff
@@ -2009,3 +2023,147 @@ def test_search_transferring_body_all_accordions_open_with_open_all_query(
 
         assert "checked" in checkbox.attrs
         assert all("open" in details.attrs for details in details_elements)
+
+    @patch("app.main.util.search_utils.OpenSearch")
+    def test_search_transferring_body_file_name_in_source_shown_if_not_in_highlight(
+        self,
+        mock_search_client,
+        client,
+        mock_standard_user,
+        browse_consignment_files,
+    ):
+        """
+        Given a standard user
+        When they make a GET request to the search page with a query
+        If the file_name is not found inside highlight
+        Then it should be shown as the value inside _source AND be a link AND be the 2nd row
+        """
+        mock_search_client.return_value = MockOpenSearch(
+            search_return_value=os_mock_return_tb
+        )
+        mock_standard_user(client, "first_body")
+
+        transferring_body_id = browse_consignment_files[
+            0
+        ].consignment.series.body.BodyId
+
+        response = client.get(
+            f"{self.route_url}/{transferring_body_id}",
+            data={"query": "test"},
+        )
+
+        assert response.status_code == 200
+        soup = BeautifulSoup(response.data, "html.parser")
+        anchors = soup.find_all("a")
+        anchors_text = []
+        for anchor in anchors:
+            anchors_text.append(anchor.get_text(strip=True))
+
+        table_body = soup.find("tbody")
+        table_rows_cell_values = get_table_rows_cell_values(table_body)
+        assert table_rows_cell_values[1] == ["File name", "fifth_file.doc"]
+        assert "fifth_file.doc" in anchors_text
+
+    @patch("app.main.util.search_utils.OpenSearch")
+    def test_search_transferring_body_highlight_file_name_prioritized_over_source(
+        self,
+        mock_search_client,
+        client,
+        mock_standard_user,
+        browse_consignment_files,
+    ):
+        """
+        Given a standard user
+        When they make a GET request to the search page with a query
+        If highlight has a file_name field
+        Then it should be shown in place of the file_name inside _source AND be an achor AND is the 2nd row
+        """
+        mock_search_client.return_value = MockOpenSearch(
+            search_return_value={
+                "hits": {
+                    "total": {
+                        "value": 1000,
+                    },
+                    "hits": [
+                        {
+                            "_source": {
+                                "file_name": "fifth_file.doc",
+                            },
+                            "highlight": {
+                                "foo": ["bar"],
+                                "marco": ["polo"],
+                                "file_name": ["<mark>test_file.pdf</mark>"],
```

Comment:
Is this an actual scenario we'd expect? I.e is this a valid output from opensearch? When would this happen?

Reply:
yes its pretty much identical to what we'd get from OS, it would happen when getting highlighting for the file_name field where its anything besides the 1st or 2nd result, here its the 3rd but we need to always show it as 2nd